travis.yml: install bashate with .deb files for Ubuntu 18.04 LTS

kakurasan opened the pull request:
The new script `travis-install-bashate.sh` installs bashate
by unpacking .deb files built for Ubuntu Bionic (18.04 LTS).
This also reduces job time.

Ubuntu Bionic will be supported until April 2023.

Also, ShellCheck is already available on Ubuntu image and
we don't need to install it.

austin987 commented on misc/travis-install-bashate.sh:
```diff
@@ -0,0 +1,65 @@
+#       - time sh ./misc/travis-install-bashate.sh
+#       - ...
+
+set -x
```
You probably want (at least) set -e and set -u as well.

kakurasan replied:
`set -x` was replaced with `set -eux`, and `if ... fi` for `curl` was removed.

austin987 commented on .travis.yml:
```diff
@@ -16,23 +16,17 @@ dist: trusty
-    - if [[ "$TRAVIS_OS_NAME" == "linux" ]] ; then echo "deb http://archive.ubuntu.com/ubuntu/ artful universe" | sudo tee -a /etc/apt/sources.list ; fi
-    - if [[ "$TRAVIS_OS_NAME" == "linux" ]] ; then time sudo apt-get -q update ; fi
-    - if [[ "$TRAVIS_OS_NAME" == "linux" ]] ; then time sudo apt-get install python-bashate shellcheck ; fi
-
```
Please add a comment about why you're installing via this script.

kakurasan replied:
Done.

austin987 commented:
What's the time difference? It seems like this will have higher overhead, since we're hardcoding version names. Maybe it should install from repos as a fallback.

kakurasan commented:
Logs in TravisCI say that `apt-get`, `easy_install`, and `pip` take longer time than this new script.

* Job 525.1 (Ubuntu):
    * `sudo apt-get -q update`: 7.15s
    * `sudo apt-get install`: 8.02s
    * bashate + shellcheck installation: 15.17s

* Job 525.2 (macOS):
    * `sudo easy_install pip`: 4.79s
    * `sudo pip install bashate`: 3.10s
    * bashate installation: 7.89s

* Job 528.1 (Ubuntu):
    * New script: 0.86s
* Job 528.2 (macOS):
    * New script: 2.14s

The total job time is not always reduced by this script.

I added a commit to skip Homebrew's `update` and this considerably reduces job time on macOS.

Also, I added a script that installs bashate via `pip`. It will be used when `travis-install-bashate-deb.sh` fails. The fallback script is also tested in TravisCI (passed).

austin987 commented:
@kakurasan sorry, just getting back around to this. I put in a quick fix for OSX that now conflicts with your changes. Could you rebae please? Sorry about that.

kakurasan commented:
Done. `misc/travis-install-bashate-pip.sh` was changed to use the updated code (`curl` and `sudo python`).
